LoRaEngine/modules/LoRaWanNetworkSrvModule/LoraTools/LoRaMessage/LoRaPayloadData.cs

```diff
@@ -102,6 +102,9 @@ public LoRaPayloadData()
         {
         }
 
+        public LoRaPayloadData(ReadOnlyMemory<byte> inputMessage) : this(inputMessage.ToArray())
```

Comment:
The [constructor that this delegates to](https://github.com/Azure/iotedge-lorawan-starterkit/pull/1157/files#diff-434a94aed910f6c38befd5bd0e31ceac7a1ea711b9ac654b3a05b0b71d1fc964L110) should be made `private` to avoid two different ways of constructing a class instance where one constructor makes a copy of the data and the other doesn't. Having two different policies could lead to subtle bugs since a `ReadOnlyMemory<>` doesn't by itself guarantee that the underlying array cannot change. It's up to the class implementation to take a copy of the input and expose that in an immutable manner.